ingest/src/test/java/com/microsoft/azure/kusto/ingest/IngestClientImplTest.java

```diff
@@ -107,12 +116,127 @@ void ingestFromStream() {
+        TableReportIngestionResult ingestionResultMock = mock(TableReportIngestionResult.class);
+
+        doReturn(ingestionResultMock).when(ingestClientSpy).ingestFromFile(any(), any());
+        doThrow(new IngestionClientException("error in resultSetToCsv"))
+                .when(ingestClientSpy)
+                .resultSetToCsv(any(), any(), anyBoolean());
+
+        ResultSetSourceInfo resultSetSourceInfo = new ResultSetSourceInfo(mock(ResultSet.class));
+
+        assertThrows(IngestionClientException.class,
+                () -> ingestClientSpy.ingestFromResultSet(resultSetSourceInfo, ingestionProperties));
+    }
+
+    @Test
+    void ingestFromResultSet_FileIngest_IngestionClientException() throws IngestionClientException, IngestionServiceException, SQLException {
+        IngestClient ingestClient = new IngestClientImpl(resourceManagerMock, azureStorageHelperMock);
+        // we need a spy to intercept the call to ingestFromFile so it wouldn't be called
+        IngestClient ingestClientSpy = spy(ingestClient);
+
+        IngestionClientException ingestionClientException = new IngestionClientException("Client exception in ingestFromFile");
+        doThrow(ingestionClientException).when(ingestClientSpy).ingestFromFile(any(), any());
+
+        ResultSet resultSet = getSampleResultSet();
+        ResultSetSourceInfo resultSetSourceInfo = new ResultSetSourceInfo(resultSet);
+
+        assertThrows(IngestionClientException.class,
+                () -> ingestClientSpy.ingestFromResultSet(resultSetSourceInfo, ingestionProperties));
+    }
+
+    @Test
+    void ingestFromResultSet_FileIngest_IngestionServiceException() throws IngestionClientException, IngestionServiceException, SQLException {
+        IngestClient ingestClient = new IngestClientImpl(resourceManagerMock, azureStorageHelperMock);
+        // we need a spy to intercept the call to ingestFromFile so it wouldn't be called
+        IngestClient ingestClientSpy = spy(ingestClient);
+
+        IngestionServiceException ingestionServiceException = new IngestionServiceException("Service exception in ingestFromFile");
+        doThrow(ingestionServiceException).when(ingestClientSpy).ingestFromFile(any(), any());
+
+        ResultSet resultSet = getSampleResultSet();
+        ResultSetSourceInfo resultSetSourceInfo = new ResultSetSourceInfo(resultSet);
+
+        assertThrows(IngestionServiceException.class,
+                () -> ingestClientSpy.ingestFromResultSet(resultSetSourceInfo, ingestionProperties));
+    }
+
+    @Test
+    void resultSetToCsv_Success() throws SQLException, IngestionClientException {
+        IngestClientImpl ingestClient = new IngestClientImpl(resourceManagerMock, azureStorageHelperMock);
+        ResultSet resultSet = getSampleResultSet();
+        StringWriter stringWriter = new StringWriter();
+
+        long numberOfCharsActual = ingestClient.resultSetToCsv(resultSet, stringWriter, false);
```

Comment:
Since you are using a sample result set using getSampleResultSet, it's better to generate the expected string the same method:
ResultSet getSampleResultSet(StringBuilder sb) 
--> and then you write the expected string based on how you generated the result test using the StringBuilder